sat/cli/bootsys/mgmt_power.py

```diff
@@ -255,37 +256,68 @@ def do_mgmt_shutdown_power(ssh_client, username, password, excluded_ncns, ncn_sh
-    # where "sat bootsys" commands are being run.
-    # TODO: Is there a better way to get the hostname of the first master node?
     try:
         other_ncns_by_role = get_and_verify_ncn_groups(excluded_ncns.union({'ncn-m001'}))
     except FatalBootsysError as err:
         LOGGER.error(f'Not proceeding with NCN power off: {err}')
         raise SystemExit(1)
 
-    other_ncns = list({ncn for role in ('managers', 'storage', 'workers')
-                       for ncn in other_ncns_by_role[role]})
-
+    # Shutdown workers
+    if other_ncns_by_role.get('workers', []):
+        worker_ncns = other_ncns_by_role['workers']
+        LOGGER.info(f'Shutting down worker NCNs: {", ".join(worker_ncns)}')
+        start_shutdown(worker_ncns, ssh_client)
+        LOGGER.info(f'Waiting up to {ncn_shutdown_timeout} seconds for worker NCNs to shut down...')
+        finish_shutdown(worker_ncns, username, password, ncn_shutdown_timeout, ipmi_timeout)
+
+    # Shutdown managers (except ncn-m001)
+    if other_ncns_by_role.get('managers', []):
+        manager_ncns = other_ncns_by_role['managers']
+        manager_ncns.remove('ncn-m001')  # Exclude ncn-m001
+        LOGGER.info(f'Shutting down manager NCNs: {", ".join(master_ncns)}')
+        start_shutdown(master_ncns, ssh_client)
+        LOGGER.info(f'Waiting up to {ncn_shutdown_timeout} seconds for manager NCNs to shut down...')
+        finish_shutdown(master_ncns, username, password, ncn_shutdown_timeout, ipmi_timeout)
+
+    # Unmap RBD on ncn-m001
+    LOGGER.info('Unmapping and unmounting RBD devices on ncn-m001')
     try:
-        with IPMIConsoleLogger(other_ncns, username, password):
-            LOGGER.info(f'Sending shutdown command to other NCNs: {", ".join(other_ncns)}')
-            start_shutdown(other_ncns, ssh_client)
-            LOGGER.info(f'Waiting up to {ncn_shutdown_timeout} seconds for other NCNs to '
-                        f'reach powered off state according to ipmitool: {", ".join(other_ncns)}.')
-            finish_shutdown(other_ncns, username, password,
-                            ncn_shutdown_timeout, ipmi_timeout)
-            LOGGER.info('Shutdown and power off of all other NCNs complete.')
-    except ConsoleLoggingError as err:
-        LOGGER.error(f'Aborting shutdown of NCNs due failure to set up NCN console logging: {err}')
-        raise SystemExit(1)
+        # Execute rbdmap unmap-all command on ncn-m001
+        command = 'rbdmap unmap-all'
+        ssh_client.connect('ncn-m001')
+        stdin, stdout, stderr = ssh_client.exec_command(command)
+        exit_status = stdout.channel.recv_exit_status()
+        if exit_status != 0:
+            LOGGER.error(
+                f'Failed to unmap RBD on ncn-m001. '
+                f'Command {command} returned non-zero exit status: {exit_status}')
+            sys.exit(1)
+        else:
+            LOGGER.info('RBD unmapped successfully on ncn-m001')
+    except (paramiko.SSHException, socket.error) as err:
```

Comment:
Unresolved reference 'paramiko'. Must be imported

Reply:
from paramiko.ssh_exception import BadHostKeyException, AuthenticationException, SSHException

it seems to be already imported Ryan in the script

Reply:
Oh, then just remove `paramiko.` here like this:

```suggestion
    except (SSHException, socket.error) as err:
```